### PR TITLE
Define env var for port mapping without host port.

### LINKED ISSUE #4005: Also create 'PORT_<NAME>' env variables for ports defined in port mappings
Right now, `PORT_<NAME>` env variables are only added for ports defined in `portDefinitions`, but not for ports defined in `portMappings`.

cc/ @jdef @jasongilanfarr

### PR DESCRIPTION
**Pull request by jeschkies**
Fixes #4005 

Is related to #4316

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/tasks/PortsMatcher.scala [hunk 1]
```diff
@@ -29,7 +29,7 @@ case class PortsMatch(hostPortsWithRole: Seq[Option[PortWithRole]]) {
 /**
   * Utility class for checking if the ports resource in an offer matches the requirements of an app.
   */
-class PortsMatcher(
+class PortsMatcher private[tasks] (
```
Comment (jdef) [inline src/main/scala/mesosphere/marathon/tasks/PortsMatcher.scala]: nit: why is the visibility modifier after the type name? doesn't it usually prefix the type declaration?

File: src/main/scala/mesosphere/mesos/TaskBuilder.scala [hunk 2]
```diff
@@ -188,7 +189,13 @@ class TaskBuilder(
...
             // The run spec uses bridge and user modes with portMappings, use them to create the Port messages
             portMappings.zip(hostPorts).collect {
-              case (portMapping, Some(hostPort)) => PortMappingSerializer.toMesosPort(portMapping, hostPort)
+              case (portMapping, None) =>
+                // No host port has been defined. See PortsMatcher.mappedPortRanges, use container port instead.
+                val updatedPortMapping =
+                  portMapping.copy(labels = portMapping.labels + ("network-scope" -> "container"))
+                PortMappingSerializer.toMesosPort(updatedPortMapping, portMapping.containerPort)
+              case (portMapping, Some(hostPort)) =>
+                PortMappingSerializer.toMesosPort(portMapping, hostPort)
```
Comment (jdef) [inline src/main/scala/mesosphere/mesos/TaskBuilder.scala]: should we also add a `network-scope` label here, with a value of `host`?

File: src/test/scala/mesosphere/mesos/TaskBuilderTest.scala [hunk 3]
```diff
@@ -1838,16 +1912,22 @@ class TaskBuilderTest extends MarathonSpec with Matchers {
...
       p.map { hostPort =>
         val b = MesosProtos.Port.newBuilder.setNumber(hostPort)
         if (name != "") b.setName(name)
         if (protocol != "") b.setProtocol(protocol)
-        if (vip != "") b.setLabels(MesosProtos.Labels.newBuilder().addLabels(
-          MesosProtos.Label.newBuilder().setKey("VIP").setValue(vip)
-        ))
+        if (labels.nonEmpty) {
+          val labelsBuilder = MesosProtos.Labels.newBuilder()
+          labels foreach { case(key, value) =>
```
Comment (jdef) [inline src/test/scala/mesosphere/mesos/TaskBuilderTest.scala]: nit: `labels.foreach`

**Comment by jdef** [comment]
minor comments. otherwise looks good

File: src/main/scala/mesosphere/marathon/tasks/PortsMatcher.scala (hunk 1 above)
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/tasks/PortsMatcher.scala]: It makes the constructor private but the class public. If the class is private the compiler complained that it was somewhere else. It's actually not an important change.

File: src/main/scala/mesosphere/mesos/TaskBuilder.scala (hunk 2 above)
Comment (jeschkies) [inline src/main/scala/mesosphere/mesos/TaskBuilder.scala]: I'm not sure. @sargun what's your suggestion?

Comment (sargun) [inline src/main/scala/mesosphere/mesos/TaskBuilder.scala]: Yeah, I need to know that as well.

**Comment by jeschkies** [comment]
retest this please

**Comment by jdef** [comment]
LGTM

**Comment by jeschkies** [comment]
Okay, I'll merge if the tests pass.

**Comment by jeschkies** [comment]
retest this please

**Comment by jdef** [comment]
:tada:

**Comment by jdef** [comment]
@aquamatthias should we be cherry-picking this for 1.3? it was tagged for that milestone...

**Comment by jasongilanfarr** [comment]
We can start 1.3.1

**Comment by jdef** [comment]
looks like it already landed in v1.3.0

**Pull request merged**